### ISSUE TITLE
Add loss registry implementation

### ISSUE THREAD

**Issue by ivan-lazarevich** [Deeplite/deeplite-torch-zoo#170]
The task is to add a registry implementation similarly like it is done for [dataset wrapper functions](https://github.com/Deeplite/deeplite-torch-zoo/pull/169/files#diff-fb90ca5409cdae9bc7d18960deee8455f261d0dc267ee707b4bad5bef66ec6ae) and [model wrapper functions](https://github.com/Deeplite/deeplite-torch-zoo/blob/c22d503e3eee73424171aaf2eaa5530ac69cf855/deeplite_torch_zoo/utils/registry.py#L69).

```
from deeplite_torch_zoo import get_loss_function

criterion = get_loss_function(dataset_name=..., model_name=...)

loss = criterion(preds, data_batch)
```

**Comment by ivan-lazarevich** [comment]
@TravisHaran
